collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java

```diff
@@ -253,8 +279,35 @@ public AuditEventLogger getAuditEventLogger() {
         return formController.getAuditEventLogger();
     }
 
-    public static class SaveResult {
+    @Override
+    public void markOriginalFileOrDelete(String questionIndex, String fileName) {
+        if (questionIndex != null && fileName != null) {
+            if (originalFiles.containsKey(questionIndex)) {
+                mediaUtils.deleteImageFileFromMediaProvider(fileName);
+            } else {
+                originalFiles.put(questionIndex, fileName);
+                stateHandle.set(ORIGINAL_FILES, originalFiles);
+            }
+        }
+    }
+
+    @Override
+    public void replaceRecentFileForQuestion(String questionIndex, String fileName) {
+        if (questionIndex != null && fileName != null) {
+            if (recentFiles.containsKey(questionIndex)) {
+                mediaUtils.deleteImageFileFromMediaProvider(recentFiles.get(questionIndex));
+            }
+            recentFiles.put(questionIndex, fileName);
+            stateHandle.set(RECENT_FILES, recentFiles);
+        }
+    }
 
+    private void clearMediaFiles() {
```

Comment:
When `saveToDisk()` task is complete, I just call `clearMediaFiles()` . So when I try to verify call to `mediaUtils.deleteImageFileFromMediaProvider()`, the test fails.

Reply:
Ah sorry because the `formSaver` is mocked! My bad. The way to verify things are set up correctly then is to assert on the files passed to the `fromSaver` instead of the call the `mediaUtils`.